lib/rules/no-invalid-regexp.js

```diff
@@ -166,8 +177,12 @@ module.exports = {
 
                     // If flags are unknown, report the regex only if its pattern is invalid both with and without the "u" flag
                     flags === null
-                        ? validateRegExpPattern(pattern, true) && validateRegExpPattern(pattern, false)
-                        : validateRegExpPattern(pattern, flags.includes("u"))
+                        ? (
+                            validateRegExpPattern(pattern, { unicode: true, unicodeSets: false }) &&
+                            validateRegExpPattern(pattern, { unicode: false, unicodeSets: true }) &&
```

Comment:
It would be good to add `valid` tests to verify that we have both checks (currently, either of these two checks can be removed and all tests will still pass),  for example: 

```js
"new RegExp('[A--B]', flags)", // valid only with `v` flag
"new RegExp('[[]\\\\u{0}*', flags)", // valid only with `u` flag
```